jax/experimental/sparse_ops.py

```diff
@@ -351,6 +360,14 @@ def _coo_fromdense_gpu_translation_rule(c, mat, *, nnz, index_dtype):
   xla.backend_specific_translations['gpu'][
       coo_fromdense_p] = _coo_fromdense_gpu_translation_rule
 
+def _coo_fromdense_jvp_rule(primals_in, tangents_in, **params):
+  mat,  = primals_in
+  mat_dot, = tangents_in
+  data, row, col = coo_fromdense(mat, **params)
+  tangents_out = ad_util.Zero.from_value(data) if type(mat_dot) is ad_util.Zero else coo_fromdense(mat_dot, **params)[0]
```

Comment:
There is a corner case here we need to worry about, I think: what if `mat_dot` has a different sparsity pattern than `mat`?

Reply:
Good point. This is not just an issue here, though, but in general. I assume in all my implementations that the primal and tangent share the same sparsity pattern - and I'm not sure how to handle the case where this isn't the point? Do you think this is something we can assume for now and work on later, or not?